### PR TITLE
improvement(cmake): simplify build option and doc

### PR DESCRIPTION
**Pull request by tpoisonooo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Build option is too complex for me.
Now that I have specified `ncnn_DIR=xx`, why have I typing `TARGET_BACKENDS=ncnn`...

If you want to use ncnn, `cmake -Dncnn_DIR=${NCNN_DIR}/build/install/lib/cmake/ncnn ..`  should be all you need. 

Please first review https://github.com/open-mmlab/mmdeploy/pull/842

## Modification

* Remove redundant build option description in doc, add a `cmake_option.md`
* Update doc

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by hanrui1sensetime** [comment]
May be still need change [jetsons.md](https://github.com/tpoisonooo/mmdeploy/blob/simplify-build-option/docs/en/01-how-to-build/jetsons.md) to remove `-DMMDEPLOY_TARGET_BACKENDS` options explicitly.
The same about zh-cn version.

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,6 +43,31 @@ if (NOT CMAKE_BUILD_TYPE)
...
+endif()
+
+if(DEFINED ncnn_DIR)
+    list(APPEND MMDEPLOY_TARGET_BACKENDS ncnn)
+endif()
+
+if(DEFINED Torch_DIR)
+    list(APPEND MMDEPLOY_TARGET_BACKENDS torchscript)
+endif()
+
```
Comment (hanrui1sensetime) [inline CMakeLists.txt:70]: Maybe need to add `InferenceEngine_DIR` for `OpenVino`?
The same as zh-cn version.

File: docs/en/01-how-to-build/android.md [hunk 2]
```diff
@@ -198,3 +129,22 @@ MMDeploy provides a recipe as shown below for building SDK with ncnn as inferenc
...
   ```
+
+Please check [cmake build option spec](cmake_option.md)
+
+#### Build Demo
+
+```Bash
+export ANDROID_ABI=arm64-v8a
+
+cd ${MMDEPLOY_DIR}/build_${ANDROID_ABI}/install/example
```
Comment (hanrui1sensetime) [inline docs/en/01-how-to-build/android.md:140]: maybe is `cd ${MMDEPLOY_DIR}/build_${ANDROID_ABI}/install/example/cpp`
The same as zh-cn version, and the demo build for other unchanged platform.

Comment (tpoisonooo) [reply docs/en/01-how-to-build/android.md]: done.

File: CMakeLists.txt (hunk 1 above)
Comment (tpoisonooo) [reply CMakeLists.txt:70]: done.

**Comment by AllentDan** [comment]
May update the building way of the dockerfile as well.

**Comment by tpoisonooo** [comment]
> May update the building way of the dockerfile as well.

done.

**Review by hanrui1sensetime — APPROVED** [review]
Please fix CircleCI first, and then LGTM.

**Comment by tpoisonooo** [comment]
> Please fix CircleCI first, and then LGTM.

done.

**Review by AllentDan — APPROVED** [review]

**Review by hanrui1sensetime — APPROVED** [review]
LGTM now.

File: docs/en/01-how-to-build/android.md [hunk 3]
```diff
@@ -198,3 +129,22 @@ MMDeploy provides a recipe as shown below for building SDK with ncnn as inferenc
 
   make -j$(nproc) && make install
   ```
+
+Please check [cmake build option spec](cmake_option.md)
+
+#### Build Demo
```
Comment (lvhan028) [inline docs/en/01-how-to-build/android.md:135]: Demos can be built by enable `-DMMDEPLOY_BUILD_EXAMPLES`, which is already presented in chapter `Build SDK Demos`. So we don't need to add a new `Build Demo` chapter any more.

File: docs/en/01-how-to-build/snpe.md [hunk 4]
```diff
@@ -136,7 +136,7 @@ $ cmake .. \
 ### 3) Compile demo
 
 ```bash
-$ cd /path/to/install/example
+$ cd /path/to/install/example/cpp
```
Comment (lvhan028) [inline docs/en/01-how-to-build/snpe.md:139]: We can remove `Compile demo` by enabling `MMDEPLOY_BUILD_EXAMPLES` in `Compile mmdeploy SDK` chapter

File: docs/zh_cn/01-how-to-build/android.md [hunk 5]
```diff
@@ -199,3 +130,22 @@ make -j$(nproc) install
 
   make -j$(nproc) && make install
   ```
+
+参考 [cmake 选项说明](cmake_option.md)
+
+#### 编译 Demo
```
Comment (lvhan028) [inline docs/zh_cn/01-how-to-build/android.md:136]: Demos can be built by enable -DMMDEPLOY_BUILD_EXAMPLES, which is already presented in chapter '编译 SDK 和 Demos'. So we don't need to add a new ‘编译 Demo’ chapter any more.

File: docs/en/01-how-to-build/cmake_option.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
+# CMake Build Option Spec
+
+<table class="docutils">
+<thead>
```
Comment (lvhan028) [inline docs/en/01-how-to-build/cmake_option.md:4]: This document should update more options, such as `MMDEPLOY_BUILD_EXAMPLES`, `MMDEPLOY_BUILD_SDK_MONOLITHIC`, `MMDEPLOY_COVERAGE`, and so on.

File: CMakeLists.txt [hunk 7]
```diff
@@ -43,6 +43,35 @@ if (NOT CMAKE_BUILD_TYPE)
...
+if(DEFINED DInferenceEngine_DIR)
```
Comment (lvhan028) [inline CMakeLists.txt:71]: `DInferenceEngine_DIR` -> `InferenceEngine_DIR`

File: CMakeLists.txt [hunk 8]
```diff
@@ -43,6 +43,35 @@ if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release CACHE STRING "choose 'Release' as default build type" FORCE)
 endif ()
 
+if(NOT CMAKE_CXX_COMPILER)
+    set(CMAKE_CXX_COMPILER g++-7)
+endif()
+
+# Fill backends with inference software directory
+if(DEFINED TENSORRT_DIR)
```
Comment (lvhan028) [inline CMakeLists.txt:51]: I found those conditions are bringing drawbacks for users who export path setter to env to set them in IDE, like Clion.
They can disable/enable backend just by changing `MMDEPLOY_TARGET_BACKENDS` without changing path setter.
image
But now, besides updating `MMDEPLOY_TARGET_BACKENDS`, they have also to delete or add the path setter

Actually, by exporting path setter to env, users don't need to pass them to make at all.
If there are multi versions, they have the ability to make up scripts and source them according to different version.
-

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**